lib/src/dev/mt_af_xdp.c

```diff
@@ -350,17 +416,24 @@ static int xdp_socket_init(struct mt_xdp_priv* xdp, struct mt_xdp_queue* xq) {
   cfg.rx_size = mt_if_nb_rx_desc(impl, port);
   cfg.tx_size = mt_if_nb_tx_desc(impl, port);
   cfg.xdp_flags = XDP_FLAGS_UPDATE_IF_NOEXIST;
+  if (!mt_is_privileged(impl)) /* this will skip load xdp prog */
```

Comment:
Why not move it to a runtime detect by checking if the server is alive or not?

Reply:
User should choose to run with root or with C-S mode